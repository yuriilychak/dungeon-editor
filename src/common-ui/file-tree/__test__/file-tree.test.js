import React from "react";
import { createMount } from "@material-ui/core/test-utils";
import FileTree from "../file-tree";

describe("FileTree test",()=> {

    const treeData = [
        { title: '.gitignore' },
        { title: 'package.json' },
        {
            title: 'src',
            isDirectory: true,
            expanded: true,
            children: [
                { title: 'styles.css' },
                { title: 'index.js' },
                { title: 'reducers.js' },
                { title: 'actions.js' },
                { title: 'utils.js' },
            ],
        },
        {
            title: 'tmp',
            isDirectory: true,
            children: [
                { title: '12214124-log' },
                { title: 'drag-disabled-file', dragDisabled: true },
            ],
        },
        {
            title: 'build',
            isDirectory: true,
            children: [{ title: 'react-sortable-tree.js' }],
        },
        {
            title: 'public',
            isDirectory: true,
        },
        {
            title: 'node_modules',
            isDirectory: true,
        },
    ];


    const onChange = jest.fn();

    it('default snapshot', () => {
        const generateNodeProps = () => ({});
        const wrapper = createMount()(
            <FileTree
                treeData={treeData}
                generateNodeProps={generateNodeProps}
                onChange={onChange}
            />
        );
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('existing icons', () => {
        const generateNodeProps = () => ({ icon: []});

        const wrapper = createMount()(
            <FileTree
                treeData={treeData}
                generateNodeProps={generateNodeProps}
                onChange={onChange}
            />
        );

        expect(wrapper.html()).toMatchSnapshot();
    });
});
