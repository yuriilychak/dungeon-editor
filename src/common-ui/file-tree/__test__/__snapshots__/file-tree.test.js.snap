// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileTree test default snapshot 1`] = `"<div style=\\"height: 300px;\\" class=\\"file-tree-root\\"><div></div></div>"`;

exports[`FileTree test existing icons 1`] = `"<div style=\\"height: 300px;\\" class=\\"file-tree-root\\"><div></div></div>"`;
